### DISCUSSION TITLE
最近很难搜出速度快的IP

### DISCUSSION THREAD

**Discussion by denniszzg** [XIU2/CloudflareSpeedTest#101] [讨论]
最近一两周，发现经常搜不出来IP，搜索条件没变。半个小时了，动都不动，之前几分钟就筛出来了。。
![Uploading 1.png…]()

**Comment by denniszzg** [comment]
![1](https://user-images.githubusercontent.com/30626214/115982529-7d5c9e80-a5ce-11eb-8be3-6c858f9edca4.png)

**Comment by XIU2** [comment]
那你尝试过不指定 延迟和速度 条件运行吗？（或者只指定 [平均延迟上限]）

**Comment by XIU2** [comment]
我感觉还行吧，刚才跑了下 `cfst_3proxy.bat` 脚本。

![01](https://user-images.githubusercontent.com/54703944/115982795-45565b00-a5d0-11eb-87b0-ff0a8b17ab83.png)

```
# 因为脚本不直接输出结果内容，所以下面的是测速结果文件内容：

IP 地址,已发送,已接收,丢包率,平均延迟,下载速度 (MB/s)
104.20.87.211,10,10,0.00,156.01,20.59
104.22.19.64,10,10,0.00,155.22,20.49
104.20.224.85,10,10,0.00,156.47,18.21
104.22.33.101,10,10,0.00,156.38,17.05
104.20.5.3,10,10,0.00,155.51,16.75
```

对于 CDN 这种东西，不同地区质量都不一样，而且是**动态变化**的。
上个月有一阵子我这里链接各个 Cloudflare CDN IP 质量也是很差（速度不说，光延迟就一直保持在 两三百ms 以上，而正常情况下白天能低至 130ms~150ms+），后来也没过多关注，不知道什么时候又正常了。

所以你也不需要太担心，或许过段时间就恢复了呢。

**Reply by denniszzg** [reply]
确实，可能性太多，很难知道背后原因，希望很快恢复正常，不然太难受了

![2](https://user-images.githubusercontent.com/30626214/115984564-4db39380-a5da-11eb-93d3-46a13169fedc.png)

**Reply by XIU2** [reply]
@denniszzg 你是什么运营商？我是联通，我 TCPing 你截图了的一些 IP，发现**大部分都是不可用（超时）**。

我平时测速结果里，大都是 **104、172** 开头的 IP（很少像你这样的），而你的测速结果里却完全没有这两个 IP 段。

我建议你新建一个文本文件（如 1.txt），写入一下内容：
```
104.16.0.0/13
104.24.0.0/14
172.64.0.0/17
172.64.128.0/18
172.64.192.0/19
172.64.224.0/22
172.64.229.0/24
172.64.230.0/23
172.64.232.0/21
172.64.240.0/21
172.64.248.0/21
172.65.0.0/16
172.67.0.0/16
```

然后运行时指定 IP 文件参数试试如何：

```
CloudflareST.exe -f 1.txt
```

**Reply by denniszzg** [reply]
我是电信。平时也是104和172，上图的结果是去掉参数跑的，不知道为什么。
按上面你的建议测试结果如下：
这是为啥呢？

![3](https://user-images.githubusercontent.com/30626214/115985998-ddf4d700-a5e0-11eb-98d0-ee3f76029be9.png)

**Reply by XIU2** [reply]
@denniszzg 不是很清楚，不过 Cloudflare CDN 分配给各域名的默认 IP 基本都是 104 和 172 这两个 IP 段（其他 IP 段比较少见，但也不是没有）。所以再找到原因之前，你可以用这个 **1.txt 替换掉 ip.txt** 去使用（文件名互换）。

**Reply by denniszzg** [reply]
刚才单独测了前三位的IP，然后又用v2rayN的测试服务器功能测速，结果如下：
波动还是蛮大的

![4](https://user-images.githubusercontent.com/30626214/115986288-15b04e80-a5e2-11eb-8936-62fbb4b2bbdf.png)
![5](https://user-images.githubusercontent.com/30626214/115986289-16e17b80-a5e2-11eb-9e67-2d608c847b6e.png)

**Reply by XIU2** [reply]
@denniszzg 同一个 IP 多次测速结果不同，这个是正常（Anycast 技术的特性？+ 毕竟是美国服务器）。

至于和代理软件测速差距大，也算是正常的。

- CloudflareST 测速：
<kbd>你 <=> Cloudflare CDN</kbd>

- 代理软件测速：
<kbd>你 <=> Cloudflare CDN <=> 代理服务器 <=> 目标网站</kbd>

中转服务器太多，每次都会损失速度，不同服务器之间的网络质量不等，所以最终测速只能低于 CloudflareST 的测试速度。

**Reply by denniszzg** [reply]
明白了:smile:

**Comment by mivtec** [comment]
正常情况下CMCC 能找出好多到HK的线路，但是会有维护的情况，维护时这些IP会route到SEA或是LAX SFO.
维护的信息可以在这里查到
[https://www.cloudflarestatus.com/](https://www.cloudflarestatus.com/)
